frontend/src/tests/user-permission-tests.js

```diff
@@ -30,26 +62,54 @@ export function userPermissionsTests(api) {
     });
 
     it("A non-admin cannot access the admin route", async () => {
-        let resp = await apiPOST("/admin/users", newUserData);
-        // Save any extra attributes we got, such as an `id`
-        Object.assign(newUserData, resp.payload);
-
-        // Set the active user to the newly-created user
-        resp = await apiPOST("/debug/active_user", { id: newUserData.id });
+        const prevUserId = await setupUserPerm(newInstructorData);
 
         // Try to fetch an admin route
-        resp = await apiGET("/admin/active_user");
+        let resp = await apiGET("/admin/active_user");
         expect(resp).toHaveStatus("error");
 
-        // Set the active_user back to the default.
-        resp = await apiPOST("/debug/active_user", {
-            id: databaseSeeder.seededData.active_user.id,
-        });
+        await restoreOriginalUserPerm(prevUserId);
+    });
+
+    it("An instructor can access instructor routes", async () => {
+        const prevUserId = await setupUserPerm(newInstructorData);
+
+        // the default user has all roles, including the instructor role
+        let resp = await apiGET("/instructor/instructors");
+        expect(resp).toHaveStatus("success");
+
+        await restoreOriginalUserPerm(prevUserId);
     });
 
-    it.todo("An instructor can access instructor routes");
-    it.todo("A TA can access TA routes");
-    it.todo(
-        "An instructor who is also a TA can access instructor and TA routes"
-    );
+    it("A TA can access TA routes", async () => {
+        // the default user has all roles, including the TA role
+        let resp = await apiGET("/ta/active_user");
+        expect(resp).toHaveStatus("success");
+
+        const prevUserId = await setupUserPerm(newTAData);
+
+        // Try to fetch an instructor route
```

Comment:
instructor -> ta

Reply:
> instructor -> ta

i do not understand this review. would you please explain this more?

Reply:
The comment says the word "instructor" but you mean to have the word "ta"